crates/client/eth/src/config.rs

```diff
@@ -0,0 +1,77 @@
+use alloy::primitives::Address;
+use serde::{Deserialize, Serialize};
+use std::fs::File;
+use std::path::PathBuf;
+
+use crate::error::Error;
+
+/// Default Anvil local endpoint
+pub const DEFAULT_RPC_ENDPOINT: &str = "http://127.0.0.1:8545";
+/// Default Anvil chain ID
+pub const DEFAULT_CHAIN_ID: u64 = 31337;
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct EthereumClientConfig {
+    #[serde(default)]
+    pub provider: EthereumProviderConfig,
+    #[serde(default)]
+    pub contracts: StarknetContracts,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum EthereumProviderConfig {
+    Http(HttpProviderConfig),
+}
+
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+pub struct StarknetContracts {
+    pub core_contract: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct HttpProviderConfig {
+    #[serde(default = "default_rpc_endpoint")]
+    pub rpc_endpoint: String,
+}
+
+fn default_rpc_endpoint() -> String {
+    DEFAULT_RPC_ENDPOINT.into()
+}
+
+fn default_chain_id() -> u64 {
+    DEFAULT_CHAIN_ID
+}
+
+impl Default for HttpProviderConfig {
+    fn default() -> Self {
+        Self { rpc_endpoint: default_rpc_endpoint() }
+    }
+}
+
+impl Default for EthereumProviderConfig {
+    fn default() -> Self {
+        Self::Http(HttpProviderConfig::default())
+    }
+}
+
+impl EthereumProviderConfig {
+    pub fn rpc_endpoint(&self) -> &String {
+        match self {
+            Self::Http(config) => &config.rpc_endpoint,
+        }
+    }
+}
+
+impl StarknetContracts {
+    pub fn core_contract(&self) -> Result<Address, Error> {
```

Comment:
core_contracts, with a `s` to be coherent with the rest

Reply:
this config rs might get deleted only, because it's not been used so far